src/widgets/table/table.rs

```diff
@@ -1484,7 +1364,264 @@ mod tests {
                 .rows(vec![])
                 .footer(Row::new(vec!["h", "i"]))
                 .column_spacing(0);
-            assert_eq!(table.get_columns_widths(10, 0), &[(0, 5), (5, 5)])
+            assert_eq!(table.get_columns_widths(10, 0), [(0, 5), (5, 5)])
+        }
+
+        #[test]
+        fn spacers_priority_unconstrained() {
+            fn test(
+                highlight_spacing: HighlightSpacing,
+                selection: Option<usize>,
+                expected: Buffer,
+            ) {
+                let table = Table::default()
+                    .rows(vec![Row::new(vec!["hello", "world"])])
+                    .highlight_spacing(highlight_spacing)
+                    .highlight_symbol(">>>")
+                    .column_spacing(0);
+                let area = Rect::new(0, 0, 15, 3);
+                let mut buf = Buffer::empty(area);
+                if let Some(s) = selection {
+                    let mut state = TableState::default().with_selected(Some(s));
+                    StatefulWidget::render(table, area, &mut buf, &mut state);
+                    assert_buffer_eq!(buf, expected);
+                } else {
+                    Widget::render(table, area, &mut buf);
+                    assert_buffer_eq!(buf, expected);
+                }
+            }
+
+            // no highlight_symbol rendered ever
+            test(
+                HighlightSpacing::Never,
+                None,
+                Buffer::with_lines(vec![
+                    "hello  world   ", // row 1
```

Comment:
why does this have two spaces if column spacing is 0?

Reply:
When widths are not given, column constraints are calculated as `vec![Constraint::Length(max_width / col_count.max(1) as u16); col_count]`, i.e. columns are split equally among available space. Here we get `Length(7)`.

This is a little odd behavior considering default layout for the table is `Flex::Start`. It would be nice to use some kind of `Proportional` constraint by default based on number of characters in each row on average. But that is out of scope for this PR and can be discussed later.

Reply:
I think it would be better to change the default to be `Flex::Stretch` here as well.

Reply:
Changing the defaults for layout type things require an proper understanding of how people are using these widgets and what and how they'd expect stability in their UI between versions. I'm generally willing to break compatibility in some things, but not without understanding the impact of that change. This seems like it would be problematic (even if it is the right thing to do)

Reply:
Regarding the tests, perhaps it would be the right thing to do to specify the widths properly so that the columns are clear and obvious. If I had to ask about why this is 2 spaces, then later when these tests break it will be difficult to understand why the 2 space or 1 space made sense.